pkg/minikube/tunnel/loadbalancer_patcher.go

```diff
@@ -0,0 +1,158 @@
+/*
+Copyright 2018 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tunnel
+
+import (
+	"fmt"
+
+	"github.com/sirupsen/logrus"
+	core_v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8s_types "k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/rest"
+	"k8s.io/minikube/pkg/minikube/tunnel/types"
+)
+
+type LoadBalancerPatcher interface {
+	PatchServices() ([]string, error)
+	Cleanup() ([]string, error)
+}
+
+type RequestSender interface {
+	Do(request *rest.Request) (result []byte, err error)
+}
+
+type PatchConverter interface {
+	Convert(restClient rest.Interface, patch *types.Patch) *rest.Request
+}
+
+type DefaultPatcher struct {
+	coreV1Client   v1.CoreV1Interface
+	requestSender  RequestSender
+	patchConverter PatchConverter
+}
+
+func (l *DefaultPatcher) PatchServices() (managedServices []string, err error) {
+	managedServices, err = l.applyOnLBServices(func(restClient rest.Interface, svc core_v1.Service) ([]byte, error) {
+		return l.updateService(restClient, svc)
+	})
+	return
+}
+
+func (l *DefaultPatcher) Cleanup() (managedServices []string, err error) {
+	managedServices, err = l.applyOnLBServices(func(restClient rest.Interface, svc core_v1.Service) ([]byte, error) {
+		return l.cleanupService(restClient, svc)
+	})
+	return
+}
+
+func (l *DefaultPatcher) applyOnLBServices(action func(restClient rest.Interface, svc core_v1.Service) ([]byte, error)) ([]string, error) {
+	services := l.coreV1Client.Services("")
+	serviceList, e := services.List(metav1.ListOptions{})
+	if e != nil {
+		return nil, e
+	}
+	restClient := l.coreV1Client.RESTClient()
+
+	var managedServices []string
+
+	for _, svc := range serviceList.Items {
+		if svc.Spec.Type == "LoadBalancer" {
```

Comment:
nit: we should add some debugging in here

Reply:
added